platform/kube/conversion.go

```diff
@@ -82,11 +82,16 @@ func convertPort(port v1.ServicePort) *model.Port {
 
 func convertService(svc v1.Service) *model.Service {
 	ports := make([]*model.Port, 0)
-	addr := ""
+	addr, external := "", false
 	if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != v1.ClusterIPNone {
 		addr = svc.Spec.ClusterIP
 	}
 
+	if svc.Spec.Type == v1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
+		addr = svc.Spec.ExternalName
```

Comment:
I think you should add a field to `Service` called `ExternalName` to hold this value. Address is needed for TCP/HTTPS routing for the in-cluster IP address.